src/lib/core/datetime.h

```diff
@@ -400,6 +401,18 @@ datetime_usec(const struct datetime *date)
 	return datetime_nsec(date) / 1000;
 }
 
+/** Simplified checker for datetime structure validity */
+static inline bool
+datetime_validate(const struct datetime *date)
+{
+	if (unlikely(date->epoch < MIN_EPOCH_SECS_VALUE) ||
```

Comment:
Is there any chance we can validate `tzindex`?

Reply:
Yup, there is known maximum for tzcode index we generated. But it's in the otehr compilation unit, so we need to invent some trick to reuse without circle dependency.

Reply:
```diff
diff --git a/src/lib/core/datetime.h b/src/lib/core/datetime.h
index 63fe45962..88ea95732 100644
--- a/src/lib/core/datetime.h
+++ b/src/lib/core/datetime.h
@@ -62,12 +62,19 @@ struct tnt_tm;
 #define MAX_TZOFFSET (14L*60)
 #define MIN_TZOFFSET (-12L*60)
 
+/**
+ * Actually we have lesser number of generated timezones, but 1024
+ * might be a good estimate.
+ */
+#define MAX_TZINDEX 1024
+
 /**
  * datetime structure keeps number of seconds and
  * nanoseconds since Unix Epoch.
@@ -420,6 +427,9 @@ datetime_validate(const struct datetime *date)
 	if (unlikely(date->tzoffset < MIN_TZOFFSET) ||
 	    unlikely(date->tzoffset > MAX_TZOFFSET))
 		return false;
+	if (unlikely(date->tzindex < 0) ||
+	    unlikely(date->tzindex > MAX_TZINDEX))
+		return false;
 	return true;
 }
 
diff --git a/src/lib/tzcode/timezone.c b/src/lib/tzcode/timezone.c
index 4f1468291..d86f4f634 100644
--- a/src/lib/tzcode/timezone.c
+++ b/src/lib/tzcode/timezone.c
@@ -25,15 +25,14 @@ static struct date_time_zone zones_raw[] = {
 #undef ZONE_ABBREV
 };
 
-#define ZONES_MAX 1024
-static_assert(ZONES_MAX >= lengthof(zones_raw), "please increase ZONES_MAX");
+static_assert(MAX_TZINDEX >= lengthof(zones_raw), "please increase MAX_TZINDEX");
 
 /**
  * Sorted array of zone descriptors, whether it's abbreviation, full
  * zone name or [backward-compatible] link name.
  */
 static struct date_time_zone zones_sorted[lengthof(zones_raw)];
-static struct date_time_zone zones_unsorted[ZONES_MAX];
+static struct date_time_zone zones_unsorted[MAX_TZINDEX];
 
 static int
 compare_zones(const void *a, const void *b)

```